### PR TITLE
Fix make will always rebuild issue.

### LINKED ISSUE #2978: make install rebuilds go binaries
**Description**

The Go binaries (`bin/crio` and `bin/crio-status`) are rebuilt when running `make install`.

**Steps to reproduce the issue:**
1. `make GIT_COMMIT=brokenbuild`
2. `bin/crio -v`
3. `mkdir /tmp/crio-bins`
4. `make install PREFIX=/tmp/crio-bins`
5. `/tmp/crio-bins/bin/crio -v`

**Describe the results you received:**

1. make rebuilds Go binaries
2. `GIT_COMMIT` setting is lost after `make install`

**Describe the results you expected:**

1. `make install` runs `install` commands only
2. `GIT_COMMIT`  (or any other setting specified during build time is preserved)

**Additional information you deem important (e.g. issue happens only occasionally):**

The workaround is to specify the same build time `make` overrides for `make install` too but the build is still sub-optimal.

### PR DESCRIPTION
**Pull request by yihuaf**
closes #2978 
**- What I did**
Fix the issue where `make` will always rebuild
**- How I did it**
1. remove the file target `bin/*` from the phony target.
2. rewrote the git commit variable in the make file to persist between runs.
**- How to verify it**
Run a clean build
**- Description for the changelog**

Fix the issue where `make` will always rebuild.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,6 +54,10 @@ GIT_VALIDATION := ${BUILD_BIN_PATH}/git-validation
 RELEASE_TOOL := ${BUILD_BIN_PATH}/release-tool
 GOLANGCI_LINT := ${BUILD_BIN_PATH}/golangci-lint
 
+COMMIT_NO := $(shell git rev-parse HEAD 2> /dev/null || true)
+GIT_COMMIT := $(if $(shell git status --porcelain --untracked-files=no),"${COMMIT_NO}-dirty","${COMMIT_NO}")
+GIT_MERGE_BASE := $(shell git merge-base origin/master $(shell git rev-parse --abbrev-ref HEAD))
```
Comment (saschagrunert) [inline Makefile]: Can we make this still work if the git executable is not available, like we did in Makefile.inc?

**Comment by TomSweeneyRedHat** [comment]
Tests are still struggling.  Code LGTM other than @saschagrunert 's comment.

**Comment by yihuaf** [comment]
```

**Comment by yihuaf** [comment]
Only see the following log from the failed e2e_fedora. There is no log on the actual test about what failed. How do i go about finding it?
```
 TASK [run e2e tests] ***********************************************************
task path: /go/src/github.com/cri-o/cri-o/contrib/test/integration/e2e.yml:32
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
########## FINISHED STAGE: FAILURE: RUN THE CRI-O E2E TESTS [02h 14m 21s] ########## 
```

**Comment by haircommander** [comment]
The kata failure doesn't seem to be related to this pr (https://github.com/cri-o/cri-o/pull/3022#issuecomment-564736297)

**Comment by yihuaf** [comment]
@haircommander I don't think the failed test has anything to do with this PR. Based on your comment, it seems to be a test issue and other PR is failing the same as well?

**Comment by TomSweeneyRedHat** [comment]
kata-jenkins is failing here too....

**Pull request merged**